vapour_api/api.py

```diff
@@ -0,0 +1,172 @@
+                FROM d_routes_position_ids
+            ) t 
+            INNER JOIN d_routes_position 
+                on t.pos = d_routes_position.id
+            WHERE rel_osm_id = :route_id
+            ORDER BY pos
+        """
+        ),
+        route_id=route_id,
+    )
+    route_stop_positions = []
+    for row in result:
+        s = {
+            "route_position_id": row[0],
+            "geojson": ast.literal_eval(row[1]),
+            "lat": row[2],
+            "lon": row[3],
+        }
+        route_stop_positions.append(s)
+    return route_stop_positions
+
+
+def get_route_stops_with_connections(route_id):
+    result = db.engine.execute(
+        text(
+            """
+            SELECT stop_osm_id, stop_osm_type, stop_index, 
+                stop_name,
+                st_asGeoJSON(ST_Transform(stop_geom, 4326)), 
+                st_X(ST_Transform(stop_geom, 4326)) as lon,
+                st_Y(ST_Transform(stop_geom, 4326)) as lat,
+                other_routes_at_stop
+            from d_route_stops_with_connections
+            where route_osm_id = :route_id
+            order by stop_index;
+        """
+        ),
+        route_id=route_id,
+    )
+    route_stops = []
+    for row in result:
+        s = {
+            "member_osm_id": row[0],
+            "member_type": row[1],
+            "name": row[3],
+            "geojson": ast.literal_eval(row[4]),
+            "lon": row[5],
+            "lat": row[6],
+            "shields": [
+                {"ref": f[1], "colour": f[2]}
+                for f in sorted(row[7], key=lambda x: x[1])
+                if f[1]
+            ],
+        }
+        route_stops.append(s)
+    return route_stops
+
+
+class Index(Resource):
+    def get(self):
+        return {"links": [{"href": "api.py/route/<route_id>"}]}
+
+
+class Route(Resource):
+    def get(self, route_id):
+        if route_id:
+            r = OrderedDict([])
+            route_properties = get_route_properties(route_id)
+            r["route_info"] = route_properties
+            route_stop_positions = get_route_stop_positions(route_id)
+            geo_stops = [Point((sp["lat"], sp["lon"])) for sp in route_stop_positions]
+            r["stop_positions"] = GeometryCollection(geo_stops)
+            route_stops = get_route_stops_with_connections(route_id)
+            for s in route_stops:
+                if s["member_type"] == 0:
+                    s["osm_id"] = "node/"
+                elif s["member_type"] == 1:
+                    s["osm_id"] = "way/"
+                else:
+                    s["osm_id"] = "relation/"
+                s["osm_id"] = s["osm_id"] + str(s["member_osm_id"])
+                s.pop("member_type")
+                s.pop("member_osm_id")
+                s.pop("geojson")
+            r["stop_list"] = route_stops
+            r["geometry"] = ast.literal_eval(get_route_geojson(route_id))
+            r["attribution"] = ATTRIBUTION
+            return r
```

Comment:
No still convinced that geojson should should the root element?

Reply:
As discussed, I think you're right about it and it is referenced in https://github.com/Jungle-Bus/VapourTrail/issues/41. I want to finish the in progress work, and them make improvements.

Reply:
Ok, let go forward.